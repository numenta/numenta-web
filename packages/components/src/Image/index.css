/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.mw {
  composes: mw-100 from 'tachyons';
}
.w {
  composes: w-100 from 'tachyons';
}


.center {
  composes: center from 'tachyons';
}
.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}
.round {
  composes: br-100 from 'tachyons';
}
.shadow {
  composes: shadow-4 from 'tachyons';
}


.image {
  composes: br2 from 'tachyons';
  composes: mb1 from 'tachyons';
}
