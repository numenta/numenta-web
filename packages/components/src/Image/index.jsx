// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import {prefixLink} from 'gatsby-helpers'
import React from 'react'

import styles from './index.css'


const Image = (props) => {
  const {
    alt, border, center, onClick, respond, round, shadow, src, title,
  } = props
  const classes = [styles.image]
  const prefix = prefixLink('')
  let source = src

  if (prefix && source && (source.indexOf(prefix) === -1)) {
    source = prefixLink(source)
  }

  if (border) classes.push(styles.border)
  if (center) classes.push(styles.center)
  if (respond) classes.push(styles[respond])
  if (round) classes.push(styles.round)
  if (shadow) classes.push(styles.shadow)

  /* eslint-disable jsx-a11y/no-static-element-interactions */
  return (
    <img
      alt={alt}
      className={classNames(...classes)}
      onClick={onClick}
      src={source}
      title={title}
    />
  )
  /* eslint-enable jsx-a11y/no-static-element-interactions */
}

Image.propTypes = {
  alt: React.PropTypes.string.isRequired,
  border: React.PropTypes.bool,
  onClick: React.PropTypes.func,
  center: React.PropTypes.bool,
  respond: React.PropTypes.oneOf(['mw', 'w']),
  round: React.PropTypes.bool,
  shadow: React.PropTypes.bool,
  src: React.PropTypes.string.isRequired,
  title: React.PropTypes.string,
}

Image.defaultProps = {
  border: false,
  center: false,
  round: false,
  shadow: true,
}

export default Image
