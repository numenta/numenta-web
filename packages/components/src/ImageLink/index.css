/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._visited:visited {
  color: #1aa0db;
}


.imagelink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: _visited;
  color: #1aa0db;
}
