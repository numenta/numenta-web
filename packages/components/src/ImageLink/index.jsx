// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import {IndexLink, Link} from 'react-router'
import {prefixLink} from 'gatsby-helpers'
import React from 'react'

import {triggerGAnalyticsEvent} from 'numenta-web-shared-utils/lib/client'

import styles from './index.css'


/**
 *
 */
const ImageLink = ({children, title, to}) => {
  const onClick = (event) => {
    // get focus off clicked image
    event.target.parentNode.blur()
    // send ga event for asset link/download
    return triggerGAnalyticsEvent(event.target.getAttribute('href'))
  }
  const attrs = {
    // default internal non-index link
    className: styles.imagelink,
    onClick,
    title,
    to: prefixLink(to),
  }
  let Node = Link

  if (to && to.match(/^.+:/)) {
    // external link (browser location)
    Node = 'a'
    attrs.href = to
    delete attrs.to
  }
  else if (to === '/') {
    // internal index main/home link
    Node = IndexLink
  }

  return (
    <Node {...attrs}>
      {children}
    </Node>
  )
}

ImageLink.propTypes = {
  children: React.PropTypes.node.isRequired,
  title: React.PropTypes.string,
  to: React.PropTypes.string.isRequired,
}

export default ImageLink
