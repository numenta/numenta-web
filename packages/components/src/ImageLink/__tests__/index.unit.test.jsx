import React from 'react'
import renderer from 'react-test-renderer'

import Image from '../../Image'
import ImageLink from '../../ImageLink'


describe('ImageLink React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <ImageLink title="Brain" to="http://numenta.com">
        <Image alt="Brain" src="brain.png" title="Brain Image" />
      </ImageLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // @TODO test different link types (see README)

})
