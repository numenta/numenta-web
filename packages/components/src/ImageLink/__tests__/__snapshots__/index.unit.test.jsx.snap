exports[`ImageLink React component Renders correctly 1`] = `
<a
  className="imagelink"
  href="http://numenta.com"
  onClick={[Function]}
  title="Brain">
  <img
    alt="Brain"
    className="image shadow"
    onClick={undefined}
    src="brain.png"
    title="Brain Image" />
</a>
`;
